src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,53 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.parser.Task;
+import duke.storage.Storage;
+import duke.taskList.TaskList;
+import duke.ui.UiText;
+
+import java.io.IOException;
+
+public class DeleteCommand extends Command {
+    public DeleteCommand(String[] msg) {
+        super(msg);
+    }
+    /**
+     * Check is the user input format correct.
+     * If is correct delete the task from taskList and the storage.
+     * @param list  TaskList
+     * @param ui    UiText
+     * @param storage   Storage
+     * @throws DukeException
+     */
+    @Override
+    public void execute(TaskList list, UiText ui, Storage storage) throws DukeException {
+        if(super.command.length == 2) {
+            try {
+                int index = Integer.parseInt(super.command[1].trim());
+                if(index > list.getList().size()) {
+                    throw new DukeException("\u1F65 OOPS! the Number you\'ve key in is to big");
+                } else if (index < 1) {
+                    throw new DukeException("OOPS!! The number should be larger than 0");
+                } else {
+                    try {
+                        Task tk = list.getList().get(index - 1);
+                        list.getList().remove(index - 1);
+                        Task.reduceByOne();
```

Comment:
I would put the reduceByOne() method in the taskList object instead of Task class. The user could perhaps have more than one task list and want to manage them seperately? But the class field can be used when counting the total number of tasks from all lists!